ddtrace/propagation/http.py

```diff
@@ -57,10 +57,12 @@ def parent_call():
 
     @staticmethod
     def extract_header_value(possible_header_names, headers, default=None):
-        for header, value in headers.items():
-            for header_name in possible_header_names:
-                if header.lower() == header_name.lower():
-                    return value
+        normalized_headers = {name.lower(): v for name, v in headers.items()}
+        for header in (_.lower() for _ in possible_header_names):
```

Comment:
as far as I can tell, this line is not necessary to run for every header, since it's a list of constants that we control (at minimum only needs to happen once per process)

Reply:
Thanks, @clutchski, that's a good point. It seems that the `POSSIBLE_` headers are only used within that module, so it would make sense to call `lower` only on the result of `get_wsgi_header` when the `frozenset`s are created.